app/perfsymboltable.cpp

```diff
@@ -170,11 +176,16 @@ QFileInfo PerfSymbolTable::findFile(const QString& path, const QString &fileName
                 return fullPath;
         }
     }
+    
+    /* we want to ensure to not check a path twice (overlap of appPath and extraPath,
+       recursive symlinks like /urs/bin/X11, ...) so we store the canonical visited paths
+       for the lookup for each library */
+    QSet<QString> checkedCanonicalPaths;
```

Comment:
reserve an arbitrary size here, ideally run it once and sample the checkedCanonicalPaths.capacity() at the end and then just add that here as a magic number. I suspect it's small, so doing something like

```
// reserve an arbitrary small number to remove the first few reallocations and rehashes
checkedCanonicalPaths.reserve(16);
```

Reply:
will do, 16 is fine